nsv13/code/modules/overmap/gravity.dm

```diff
@@ -0,0 +1,100 @@
+/* Gravity component, for all objects that you want to have a gravitational effect on overmap objects.
+This was made a component because overmap anomalies and overmap objects have two completely different inheritances, because of somebody *cough*
+
+grav_maker is added to things that should exert gravity, and the _modifier parameter affects how strong their gravitational effect is. 1 being on part with that of stars, and so forth,
+
+grav_taker is added to things that should be affected by gravity, and the _modifier parameter affects how influenced they are by gravity. for the most part this should be 1, but setting it to 0.5
+for example would be a nice upgrade possibly to a ship that makes it easier to pilot in gravity fields
+*/
+
+GLOBAL_LIST_EMPTY(grav_makers)
+GLOBAL_LIST_EMPTY(grav_takers)
+
+/datum/component/grav_maker
+	var/strength = 1
+	var/last_process = 0
+
+/datum/component/grav_maker/Initialize(_modifier = 1)
+	strength = _modifier
+
+	GLOB.grav_makers += src
+	START_PROCESSING(SSovermap, src)
+
+/datum/component/grav_maker/process()
+	var/time = min(world.time - last_process, 10)
+	time /= 10 // fuck off deciseconds
+
+	last_process = world.time
+
+	var/used_pos_src = null
+	var/used_pos_rec = null
+
+	var/obj/_parent = parent
+	if(!parent)
+		return qdel(src) //we go bye bye if we have no parent we're attatched to TODO: Make this actually work. I don't think this actually works and we get grav_makers attatched to nothing that runtime
+
+	if(!istype(/obj/structure/overmap, parent)) //make a position vector for us if we don't already have one (e.g. anomalies)
+		used_pos_src = new /datum/vector2d(_parent.x*32,_parent.y*32)
+	else
+		var/obj/structure/overmap/O = parent
+		used_pos_src = O.position //otherwise use the one we already have
+
+	for(var/datum/component/grav_taker/G in GLOB.grav_takers)
+		var/obj/G_parent = G.parent
+		if(!G_parent)
+			return qdel(G) //stop those pesky runtimes!
+		if (G_parent.z != _parent.z) //only affect grav_takers on our z-level
+			continue
+
+		if(!istype(/obj/structure/overmap, G.parent)) //make a position vector for them if they don't already have one (e.g. anomalies)
```

Comment:
```suggestion
		if(!istype(G.parent, /obj/structure/overmap)) //make a position vector for them if they don't already have one (e.g. anomalies)
```

Reply:
:eyes: wot does dat do

Reply:
You have the operator order wrong, subject goes first.